packages/storage/amplify_storage_s3/example/integration_test/download_data_test.dart

```diff
@@ -0,0 +1,261 @@
+        final stringPath = 'public/download-data-string-${uuid()}';
+        await Amplify.Storage.uploadData(
+          path: StoragePath.fromString(stringPath),
+          data: HttpPayload.string(string),
+        ).result;
+
+        addTearDown(
+          () =>
+              Amplify.Storage.remove(path: StoragePath.fromString(stringPath)),
+        );
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(stringPath),
+        ).result;
+
+        expect(downloadResult.bytes, string.codeUnits);
+      });
+
+      test('empty', () async {
+        final emptyPath = 'public/download-data-empty-${uuid()}';
+        await Amplify.Storage.uploadData(
+          path: StoragePath.fromString(emptyPath),
+          data: const HttpPayload.empty(),
+        ).result;
+
+        addTearDown(
+          () => Amplify.Storage.remove(path: StoragePath.fromString(emptyPath)),
+        );
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(emptyPath),
+        ).result;
+
+        expect(downloadResult.bytes, <int>[]);
+      });
+
+      test('stream', () async {
+        final data = [1, 2, 3];
+        final stream = Stream<List<int>>.value(data);
+
+        final streamPath = 'public/download-data-stream-${uuid()}';
+        await Amplify.Storage.uploadData(
+          path: StoragePath.fromString(streamPath),
+          data: HttpPayload.streaming(stream),
+        ).result;
+
+        addTearDown(
+          () =>
+              Amplify.Storage.remove(path: StoragePath.fromString(streamPath)),
+        );
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(streamPath),
+        ).result;
+
+        expect(downloadResult.bytes, data);
+      });
+
+      test('with identity ID', () async {
+        final userIdentityId = await signInNewUser();
+        final name = 'upload-data-with-identity-id-${uuid()}';
+        final data = 'with identity ID'.codeUnits;
+        final expectedResolvedPath = 'private/$userIdentityId/$name';
+        addTearDown(
+          () => Amplify.Storage.remove(
+            path: StoragePath.fromString(expectedResolvedPath),
+          ),
+        );
+        await Amplify.Storage.uploadData(
+          data: HttpPayload.bytes(data),
+          path: StoragePath.fromIdentityId(
+            (identityId) => 'private/$identityId/$name',
+          ),
+        ).result;
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(expectedResolvedPath),
+        ).result;
+        expect(downloadResult.bytes, data);
+      });
+    });
+
+    group('with options', () {
+      test('getProperties', () async {
+        // Upload a file with properties
+        final path = 'public/download-data-get-properties-${uuid()}';
+
+        const metadata = {'foo': 'bar'};
+
+        await Amplify.Storage.uploadData(
```

Comment:
todo: use the file uploaded in the setup

We do not need to exercise the public API option for upload in these tests. There will be separate tests for upload.

Reply:
Apply this change to downloadFile as well